loadtest/control/actions.go

```diff
@@ -471,6 +471,40 @@ func SearchUsers(u user.User) UserActionResponse {
 	})
 }
 
+// AutoCompleteUsers returns "user found" error when the auto completion is
+// done successfully. In that case UserActionResponse.Info carries the user
+// name information.
+func AutoCompleteUsers(u user.User) UserActionResponse {
+	channel, err := u.Store().CurrentChannel()
+	if err != nil {
+		return UserActionResponse{Err: err}
+	}
+
+	user, err := u.Store().RandomUser()
+	if err != nil {
+		return UserActionResponse{Err: err}
+	}
+
+	cutoff := 2 + rand.Intn(len(user.Username)/2)
+
+	return emulateUserTyping(user.Username, func(term string) UserActionResponse {
+		users, err := u.AutoCompleteUsersInChannel(channel.TeamId, channel.Id, term, 100)
+		if err != nil {
+			return UserActionResponse{Err: err}
+		}
+
+		if len(term) == cutoff {
+			return UserActionResponse{Err: errors.New("user found"), Info: user.Username}
+		}
+
+		if _, ok := users[term]; ok {
```

Comment:
I am wondering if there can be any case where if the term is found, it does not satisfy the cutoff but matches the map entry ? Since cutoff is smaller than the username, it should match earlier. And if it does not match, both will fail.

Reply:
Good point 👍 Actually after a second thought, we can check the `len(users) == 1` and remove this one.